App.tsx

```diff
@@ -500,6 +500,13 @@ export const App: React.FC<IProps> = (): React.ReactElement => {
                 </Button>
 
                 <FilterDrawer
+                  onRequestClose={() => {
+                    if (isFilterDrawerSecondaryViewActive) {
+                      setIsFilterDrawerSecondaryViewActive(false);
+                    } else {
+                      setIsFilterDrawerOpen(false);
+                    }
+                  }}
```

Comment:
I think this logic should be internal to the FilterDrawerModal.

Reply:
Actually the states for showing drawer and secondary content are managed outside of this component. That's why I put this logic here.

Reply:
got it